### ISSUE TITLE
Python library fixes tool old version of typing-extensions

### ISSUE THREAD

**Issue by zlobober** [ytsaurus/ytsaurus#187]
I need to use pydantic simultaneously with ytsaurus-client. Trying to install both of them from the pip fails with the following resolution:

```
o_O max@precision []: ~ $ pip3 install pydantic --bre
Defaulting to user installation because normal site-packages is not writeable
Collecting pydantic
  Obtaining dependency information for pydantic from https://files.pythonhosted.org/packages/0a/2b/64066de1c4cf3d4ed623beeb3bbf3f8d0cc26661f1e7d180ec5eb66b75a5/pydantic-2.5.2-py3-none-any.whl.metadata
  Downloading pydantic-2.5.2-py3-none-any.whl.metadata (65 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 65.2/65.2 kB 2.4 MB/s eta 0:00:00
Collecting annotated-types>=0.4.0 (from pydantic)
  Obtaining dependency information for annotated-types>=0.4.0 from https://files.pythonhosted.org/packages/28/78/d31230046e58c207284c6b2c4e8d96e6d3cb4e52354721b944d3e1ee4aa5/annotated_types-0.6.0-py3-none-any.whl.metadata
  Downloading annotated_types-0.6.0-py3-none-any.whl.metadata (12 kB)
Collecting pydantic-core==2.14.5 (from pydantic)
  Obtaining dependency information for pydantic-core==2.14.5 from https://files.pythonhosted.org/packages/bf/ed/ee221482b51f368884ea6453f3784eeaeb17f5b737589d39d68a89bffde7/pydantic_core-2.14.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata
  Downloading pydantic_core-2.14.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.5 kB)
Collecting typing-extensions>=4.6.1 (from pydantic)
  Obtaining dependency information for typing-extensions>=4.6.1 from https://files.pythonhosted.org/packages/24/21/7d397a4b7934ff4028987914ac1044d3b7d52712f30e2ac7a2ae5bc86dd0/typing_extensions-4.8.0-py3-none-any.whl.metadata
  Downloading typing_extensions-4.8.0-py3-none-any.whl.metadata (3.0 kB)
Downloading pydantic-2.5.2-py3-none-any.whl (381 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 381.9/381.9 kB 15.6 MB/s eta 0:00:00
Downloading pydantic_core-2.14.5-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.1 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 16.4 MB/s eta 0:00:00
Downloading annotated_types-0.6.0-py3-none-any.whl (12 kB)
Downloading typing_extensions-4.8.0-py3-none-any.whl (31 kB)
Installing collected packages: typing-extensions, annotated-types, pydantic-core, pydantic
  Attempting uninstall: typing-extensions
    Found existing installation: typing_extensions 4.5.0
    Uninstalling typing_extensions-4.5.0:
      Successfully uninstalled typing_extensions-4.5.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
ytsaurus-client 0.13.7 requires typing-extensions==4.5.0, but you have typing-extensions 4.8.0 which is incompatible.
Successfully installed annotated-types-0.6.0 pydantic-2.5.2 pydantic-core-2.14.5 typing-extensions-4.8.0
```

Is it possible to bump the required version for ytsaurus-client, or (even better) make ytsaurus-client require version >= some? I'd expect all restrictions to be of form >= by default, and be switched to == case-by-case basis when some library actually performs a backward-incompatible change.

**Comment by Kontakter** [comment]
> be switched to == case-by-case basis when some library actually performs a backward-incompatible change.

Unfortunately it gives bad experience for the user: sometimes we will observe the situation that ytsaurus-client installed and it doesn't work because some library comes with new version and breaks some compatibility. It was one of the reasons in ancient times of bundling all the dependencies in yt.packages

In your case I think that you should use the older version of pydantic, or use separate virtual envs for ytsaurus-client and pydantic. Also we will update the versions in ytsaurus-client dependencies with the next update, it also will resolve your issue.

**Comment by zlobober** [comment]
I cannot use separate versions because I need to use ytsaurus-client and pydantic within a single script.

I do have a workaround of fixing the older version of pydantic, and I already used it. Still, bumping these versions in some future should eventually happen, we do not want to get stuck with old dependencies forever. Consider this issue as a request to bump the version of typing-extensions in particular to something fresh.

**Comment by zlobober** [comment]
Bundling this dependency within ytsaurus-client also works, btw.

**Comment by denvr** [comment]
We will check and update versions in the next release (still with strict dependencies ("==") for reasons of stability). And then we will update them automatically.

And it seems to be a good idea to provide weaker dependencies (">=") and strict ones at the same time

**Comment by zlobober** [comment]
I did not get the last paragraph, it looks contradicting to the first one. So are you going to eventually switch to weaker dependencies, or not?

**Comment by Kontakter** [comment]
I have fixed it, try `ytsaurus-client==0.13.9`
